c/src/test/java/org/apache/arrow/c/ExceptionTest.java

```diff
@@ -0,0 +1,184 @@
+import org.apache.arrow.vector.VectorLoader;
+import org.apache.arrow.vector.VectorSchemaRoot;
+import org.apache.arrow.vector.VectorUnloader;
+import org.apache.arrow.vector.dictionary.Dictionary;
+import org.apache.arrow.vector.dictionary.DictionaryProvider;
+import org.apache.arrow.vector.ipc.ArrowReader;
+import org.apache.arrow.vector.ipc.message.ArrowRecordBatch;
+import org.apache.arrow.vector.types.pojo.ArrowType;
+import org.apache.arrow.vector.types.pojo.Field;
+import org.apache.arrow.vector.types.pojo.Schema;
+import org.junit.jupiter.api.Test;
+
+// Regression test for https://github.com/apache/arrow-java/issues/759
+final class ExceptionTest {
+  @Test
+  public void testException() throws IOException {
+    final Schema schema =
+        new Schema(Collections.singletonList(Field.nullable("ints", new ArrowType.Int(32, true))));
+    final List<Object> batches = new ArrayList<>();
+
+    try (BufferAllocator allocator = new RootAllocator();
+         VectorSchemaRoot root = VectorSchemaRoot.create(schema, allocator)) {
+      final IntVector ints = (IntVector) root.getVector(0);
+      VectorUnloader unloader = new VectorUnloader(root);
+
+      root.allocateNew();
+      ints.setSafe(0, 1);
+      ints.setSafe(1, 2);
+      ints.setSafe(2, 4);
+      ints.setSafe(3, 8);
+      root.setRowCount(4);
+      batches.add(unloader.getRecordBatch());
+
+      final String exceptionMessage = "This is a message for testing exception.";
+
+      RuntimeException exToThrow = new RuntimeException(exceptionMessage);
+      batches.add(exToThrow);
+
+      StringWriter sw = new StringWriter();
+      PrintWriter pw = new PrintWriter(sw);
+      exToThrow.printStackTrace(pw);
+      final String expectExceptionMessage = sw.toString();
+
+      ArrowReader source = new ExceptionMemoryArrowReader(allocator, schema, batches);
+
+      try (final ArrowArrayStream stream = ArrowArrayStream.allocateNew(allocator);
+           final VectorSchemaRoot importRoot = VectorSchemaRoot.create(schema, allocator)) {
+        final VectorLoader loader = new VectorLoader(importRoot);
+        Data.exportArrayStream(allocator, source, stream);
+
+        Throwable exceptionThrowed = null;
+        try (final ArrowReader reader = Data.importArrayStream(allocator, stream)) {
+          assertThat(reader.getVectorSchemaRoot().getSchema()).isEqualTo(schema);
+
+          for (Object batch : batches) {
+            try {
+              reader.loadNextBatch();
+            } catch (Exception e) {
+              assertThat(exceptionThrowed).isEqualTo(null);
+              final String eMessage = e.getMessage();
+              // 1 for '}', ref to CDataJniException
+              assertThat(eMessage.length()).isGreaterThan(expectExceptionMessage.length() + 1); 
+              assertThat(eMessage.substring(eMessage.length() - expectExceptionMessage.length() - 1, eMessage.length() - 1))
+                  .isEqualTo(expectExceptionMessage);
```

Comment:
Just use `contains`...

Reply:
No, we must make sure the string in the exception is the same with we throws.
"abc\x037" contains "abc", but it's not correct and we want to catch the bug.

Reply:
No.
Here is the CDataJniException::getMessage:
```java
public String getMessage() {
        return "CDataJniException{errno=" + this.errno + ", message=" + super.getMessage() + '}';
    }
```
The message of the exception we thrown is `super().getMessage()`.